exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/IncomingBuffers.java

```diff
@@ -129,6 +134,10 @@ public int getRemainingRequired() {
     return collectorMap.get(senderMajorFragmentId).getBuffers();
   }
 
+  public void setOprAllocator(int senderMajorFragmentId, BufferAllocator oprAllocator) {
```

Comment:
Consider introducing `getCollector(int senderMajorFragmentId)` instead of `setOprAllocator` and `getBuffers`;

Reply:
I feel the Collector is an internal class (also used for synchronization). Why exposed it if this is not necessary?

Reply:
`DataCollector` is a top-level interface.

Reply:
I don't mind doing so if you think Collector is a top-level interface. I wish there is a better way to indicate which class is an internal framework class vs a public one safe for all operators to access. If you look closely at the code, the DataCollector interface was created to allow multiple implementations; the Drill code exposed the RawFragmentBatchProvider as the main IF for operators to consume incoming data.

Reply:
It can be done using annotations (I do not suggest to introduce such annotations now).

I don't see a need to hide DataCollector as long as HashMap is not exposed directly or concrete type of a class that implements DataCollector is exposed.